### PR TITLE
test with another  address

### PR DESCRIPTION
**Pull request by lexy212**
# Description of the change

The getAddress details test fails when it's tested with this address `0x176366cFD97885245fAEA72f8cB6951e52655Adf`. The types break with the error below. I believe this  is due to the fact that some expected fields are undefined. 

```sh
  ● BlockbookEthereum › getAddressDetails › succeeds

    TypeError: Invalid type - Expected type number for (Paginated & AddressDetailsEthereumTxids).nonTokenTxs, but got: undefined

      127 |       return value as T
      128 |     }
    > 129 |     return assertType(codec, value, ...rest)
          |            ^
      130 |   }
```

The address details looks like this via blockbook api https://eth1.trezor.io/api/address/0x176366cFD97885245fAEA72f8cB6951e52655Adf

```ts
{
page: 1,
totalPages: 1,
itemsOnPage: 1000,
address: "0x176366cFD97885245fAEA72f8cB6951e52655Adf",
balance: "0",
unconfirmedBalance: "0",
unconfirmedTxs: 0,
txs: 0,
nonce: "0"
}
```

### CODE REVIEW

**Comment by lexy212** [comment]
@dylanseago this is a simulation with no intention of merging. 

It's just the reason why this other PR https://github.com/bitaccess/blockbook-client/pull/27 had those changes.

**Comment by dylanseago** [comment]
> @dylanseago this is a simulation with no intention of merging.
> 
> It's just the reason why this other PR #27 had those changes.

@lexy212 This is great, why not keep this test? It shows that there's an edge case that breaks the current typings and needs to be fixed. Regarding the fix, it seems like only `nonTokenTxs` needs to be changed to optional? Can everything in `TokenDetailsERC20` be kept required?

File: test/BlockbookEthereum.test.ts [hunk 1]
```diff
@@ -31,11 +32,24 @@ describe('BlockbookEthereum', () => {
...
+    it('succeeds', async () => {
+      expect(await bb.getAddressDetails('0x176366cFD97885245fAEA72f8cB6951e52655Adf')).toBeDefined()
+    })
+  })
+
+  describe('getAddressDetails - faa.st wallet', () => {
+    it('succeeds', async () => {
+      expect(await bb.getAddressDetails('0x94fe3ad91dacba8ec4b82f56ff7c122181f1535d')).toBeDefined()
+    })
+  })
```
Comment (lexy212) [inline test/BlockbookEthereum.test.ts:52]: @dylanseago I was able to simulate the other breaking test.

```ts

  ● BlockbookEthereum › getAddressDetails - faa.st wallet › succeeds

    TypeError: Invalid type - Expected type "ERC20" | undefined for (Paginated & AddressDetailsEthereumTxids).tokens.type, but got: undefined

      127 |       return value as T
      128 |     }
    > 129 |     return assertType(codec, value, ...rest)
          |            ^
      130 |   }
      131 |
      132 |
```

Comment (dylanseago) [reply test/BlockbookEthereum.test.ts:52]: @lexy212 I looked at the raw response we're receiving and couldn't find any token where `type` is not `'ERC20'` using (`tokens.find((t) => t.type !== 'ERC20')`).

Are you able to identify that the data is in fact bad or is there some other sneaky bug causing this?

File: src/types/ethereum.ts [hunk 2]
```diff
@@ -80,14 +80,14 @@ export const TokenDetailsERC20 = t.type({
...
-  symbol: t.string, // 'CEN',
+  symbol: optional(t.string), // 'CEN',
 }, 'TokenDetailsERC20')
 export type TokenDetailsERC20 = t.TypeOf<typeof TokenDetailsERC20>
 
 export const TokenDetailsERC20Balance = extendCodec(
   TokenDetailsERC20,
   {
-    balance: t.string, // '8503600000000000000'
+    balance: optional(t.string), // '8503600000000000000'
```
Comment (lexy212) [inline src/types/ethereum.ts:90]: Symbol and balance was undefined here: 

```
  {
         "type": "ERC20",
         "name": "0xE0B7927c4aF23765Cb51314A0E0521A9645F0E2A",
         "contract": "0xE0B7927c4aF23765Cb51314A0E0521A9645F0E2A",
         "transfers": 118
      }
```  

● BlockbookEthereum › getAddressDetails - [faa.st](http://faa.st/) wallet › succeeds

    TypeError: Invalid type - Expected type string | undefined for (Paginated & AddressDetailsEthereumTxids).tokens.16.symbol, but got: undefined

      127 |       return value as T
      128 |     }
    > 129 |     return assertType(codec, value, ...rest)
          |            ^
      130 |   }
      131 |
      132 |

      ✕ succeeds (2117 ms)

  ● BlockbookEthereum › getAddressDetails - [faa.st](http://faa.st/) wallet › succeeds

    TypeError: Invalid type - Expected type string | undefined for (Paginated & AddressDetailsEthereumTxids).tokens.16.balance, but got: undefined

      127 |       return value as T
      128 |     }
    > 129 |     return assertType(codec, value, ...rest)
          |            ^
      130 |   }
      131 |

File: src/types/ethereum.ts [hunk 3]
```diff
@@ -107,7 +107,7 @@ export const AddressDetailsEthereumTokens = extendCodec(
   AddressDetailsEthereumBasic,
   {},
   {
-    tokens: TokenDetailsERC20,
+    tokens: t.array(TokenDetailsERC20),
```
Comment (lexy212) [inline src/types/ethereum.ts:110]: This was the major culprit @dylanseago , the data comes back as an array in the blockbook response, but our types are setup to look for an object.

File: src/types/ethereum.ts [hunk 4]
```diff
@@ -117,7 +117,7 @@ export const AddressDetailsEthereumTokenBalances = extendCodec(
   AddressDetailsEthereumBasic,
   {},
   {
-    tokens: TokenDetailsERC20Balance,
+    tokens: t.array(TokenDetailsERC20Balance),
```
Comment (lexy212) [inline src/types/ethereum.ts:120]: same thing as above

File: src/types/ethereum.ts (hunk 2 above)
Comment (lexy212) [reply src/types/ethereum.ts:90]: Project was dissolved. see etherscan screeshot

Screenshot 2022-03-28 at 23 34 28

File: src/types/ethereum.ts (hunk 3 above)
Comment (dylanseago) [reply src/types/ethereum.ts:110]: Wow! Yep. That'd do it. Great work finding this.

**Review by dylanseago — APPROVED** [review]
